### PR TITLE
DOC: fix doc build warnings

### PR DESCRIPTION
**Pull request by jreback**

### CODE REVIEW

**Comment by jreback** [comment]
cc @jorisvandenbossche

File: doc/source/indexing.rst [hunk 1]
```diff
@@ -1625,6 +1625,7 @@ This is the correct access method
 This *can* work at times, but is not guaranteed, and so should be avoided
 
 .. ipython:: python
+   :okexcept:
```
Comment (jorisvandenbossche) [inline doc/source/indexing.rst]: I think this should be an `:okwarning:` ?

Comment (jreback) [inline doc/source/indexing.rst]: right

File: doc/source/whatsnew/v0.18.0.txt [hunk 2]
```diff
@@ -780,7 +780,7 @@ Now, you can write ``.resample`` as a 2-stage operation like groupby, which
 yields a ``Resampler``.
 
 .. ipython:: python
-
+   :okwarning:
 
    r = df.resample('2s')
    r
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.18.0.txt]: This should not give a warning?

File: doc/source/whatsnew/v0.18.0.txt [hunk 3]
```diff
@@ -870,6 +870,7 @@ In prior versions, new columns assignments in an ``eval`` expression resulted
 in an inplace change to the ``DataFrame``. (:issue:`9297`)
 
 .. ipython:: python
+   :okwarning:
 
    df = pd.DataFrame({'a': np.linspace(0, 10, 5), 'b': range(5)})
    df.eval('c = a + b')
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.18.0.txt]: This is about previous behaviour, so maybe use  code-block? (the warning should not be there, as this is introduced in 0.18)

Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: will fix

File: doc/source/whatsnew/v0.9.1.txt [hunk 4]
```diff
@@ -89,12 +89,21 @@ New features
...
+                        parse_cols='A:D')
+       Out[5]:
+                          A         B         C
+       2000-01-03  0.980269  3.685731 -0.364217
+       2000-01-04  1.047916 -0.041232 -0.161812
+       2000-01-05  0.498581  0.731168 -0.537677
+       2000-01-06  1.120202  1.567621  0.003641
+       2000-01-07 -0.487094  0.571455 -1.611639
+       2000-01-10  0.836649  0.246462  0.588543
+       2000-01-11 -0.157161  1.340307  1.195778
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.9.1.txt]: Can you leave this out? As it is IMO actually a regression that `parse_dates` does not work anymore (ref https://github.com/pydata/pandas/pull/12051)

File: doc/source/whatsnew/v0.15.1.txt [hunk 5]
```diff
@@ -110,19 +110,22 @@ API changes
...
-  .. ipython:: python
-
-    s.loc[3.5:1.5]
+  .. code-block:: python
 
+     In [2]: s.loc[3.5:1.5]
+     Out[2]:
+     3    b
+     2    c
+     dtype: object
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.15.1.txt]: This should not be needed (when #12370 is merged) I think? So would leave this out as well

File: doc/source/whatsnew/v0.9.1.txt (hunk 4 above)
Comment (jreback) [inline doc/source/whatsnew/v0.9.1.txt]: done

File: doc/source/whatsnew/v0.15.1.txt (hunk 5 above)
Comment (jreback) [inline doc/source/whatsnew/v0.15.1.txt]: doen

**Comment by jreback** [comment]
so this looks reasonable.

@jorisvandenbossche

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request closed without merging**